### PR TITLE
Commits tide

### PR DESCRIPTION
**Pull request by dlibenzi**

### CODE REVIEW

File: torch_xla/csrc/torch_util.h [hunk 1]
```diff
@@ -15,4 +16,9 @@ XlaModule::TensorBatchVector XlaCreateTensorList(const py::tuple& tuple);
 // Packs a vector of XLA tensors into a Python tuple, if they are more than one.
 py::object XlaPackTensorList(const XlaModule::TensorBatchVector& outputs);
 
+// Makes a depp copy of an ATEN tensor.
```
Comment (asuhan) [inline torch_xla/csrc/torch_util.h]: depp -> deep

File: scripts/gen.py [hunk 2]
```diff
@@ -597,6 +599,14 @@ def get_return_type_str(t, orig_sig):
   return orig_sig[0:token.column - 2]
 
 
+def generate_debug_code(t, fname, rname, params, param_vars, ref_param):
```
Comment (asuhan) [inline scripts/gen.py]: This isn't really debug, it's metrics. Maybe `generate_metrics_code`?

File: torch_xla/csrc/ir.h [hunk 3]
```diff
@@ -59,7 +59,7 @@ struct Output {
   struct Hasher {
     size_t operator()(const Output& output) const {
       size_t h = reinterpret_cast<std::ptrdiff_t>(output.node);
-      return h ^ (h >> 11) ^ output.index;
+      return h ^ (output.index + 0x9e3779b97f4a7c15 + (h << 6) + (h >> 2));
```
Comment (asuhan) [inline torch_xla/csrc/ir.h]: Any reason why we couldn't use `std::hash`?

File: scripts/gen.py (hunk 2 above)
Comment (dlibenzi) [reply scripts/gen.py]: It is supposed to generate VLOG debug as well. IMO the name is OK. After all even the metrics are a debugging tool.

File: torch_xla/csrc/ir.h (hunk 3 above)
Comment (dlibenzi) [reply torch_xla/csrc/ir.h]: We are combining here.
This is the same "combine" code in utill.h, but cannot include util.h in this file for layering issues.

**Pull request merged**